pubsub/gcloud/aio/pubsub/subscription.py

```diff
@@ -1,66 +1,106 @@
 import asyncio
-import logging
+import concurrent.futures
+import signal
+from asyncio import Future  # pylint: disable=ungrouped-imports
+from typing import Any
+from typing import Callable
+from typing import Optional
 
-import google.auth.exceptions
-import google.cloud.pubsub as pubsub
-import google.gax.errors
-import grpc
+from google.api_core import exceptions
+from google.cloud import pubsub
+from google.cloud.pubsub_v1.subscriber.message import Message
+from google.cloud.pubsub_v1.types import FlowControl
+from google.cloud.pubsub_v1.types import Scheduler
 
+from .utils import convert_google_future_to_concurrent_future
 
-log = logging.getLogger(__name__)
 
+class Subscription:
 
-class Subscription(pubsub.subscription.Subscription):
-    async def acknowledge(self, ack_ids, client=None, retries=0):
-        try:
-            return super().acknowledge(ack_ids, client=client)
-        except (google.auth.exceptions.TransportError,
-                google.gax.errors.RetryError):
-            # common intermittent communication errors
-            if not retries:
-                raise
-
-            await asyncio.sleep(1)
-
-        return await self.acknowledge(ack_ids, client=client,
-                                      retries=retries-1)
+    def __init__(self,
+                 subscription: str,
+                 *,
+                 loop: Optional[asyncio.AbstractEventLoop] = None
+                 ) -> None:
+        self.subscription = subscription
```

Comment:
Looks like the Google API has this value passed in to `create_subscription` and `subscribe` rather than setting it in `init` -- let's try to follow their interface here to make this as close to a drop-in as possible.